sui/deploy-contract.js

```diff
@@ -5,156 +5,206 @@ const { toB64 } = require('@mysten/sui/utils');
-                `${packageId}::gas_service::GasCollectorCap`,
-            ]);
-            contractConfig.objects = { GasService, GasCollectorCap };
-            break;
-        }
-
-        case 'axelar_gateway': {
-            const { minimumRotationDelay, domainSeparator, policy, previousSigners } = options;
-            const operator = options.operator || keypair.toSuiAddress();
-            const signers = await getSigners(keypair, config, chain, options);
-
-            validateParameters({ isNonEmptyString: { previousSigners, minimumRotationDelay }, isKeccak256Hash: { domainSeparator } });
-
-            const [creatorCap, relayerDiscovery, upgradeCap] = getObjectIdsByObjectTypes(publishTxn, [
-                `${packageId}::gateway::CreatorCap`,
-                `${packageId}::discovery::RelayerDiscovery`,
-                `${suiPackageAddress}::package::UpgradeCap`,
-            ]);
+    printInfo('Register transaction', registerTx.digest);
+}
 
-            const encodedSigners = signersStruct
-                .serialize({
-                    ...signers,
-                    nonce: bytes32Struct.serialize(signers.nonce).toBytes(),
-                })
-                .toBytes();
+async function postDeployAxelarGateway(published, keypair, client, config, chain, options) {
+    const { packageId, publishTxn } = published;
+    const { minimumRotationDelay, policy, previousSigners } = options;
+    const operator = options.operator || keypair.toSuiAddress();
+    const signers = await getSigners(keypair, config, chain, options);
+    const domainSeparator = await getDomainSeparator(config, chain, options);
+
+    validateParameters({
+        isNonEmptyString: { previousSigners },
+        isValidNumber: { minimumRotationDelay },
+    });
+
+    const [creatorCap, relayerDiscovery, upgradeCap] = getObjectIdsByObjectTypes(publishTxn, [
+        `${packageId}::gateway::CreatorCap`,
+        `${packageId}::discovery::RelayerDiscovery`,
+        `${suiPackageAddress}::package::UpgradeCap`,
+    ]);
+
+    const encodedSigners = signersStruct
+        .serialize({
+            ...signers,
+            nonce: bytes32Struct.serialize(signers.nonce).toBytes(),
+        })
+        .toBytes();
+
+    const tx = new Transaction();
+
+    const separator = tx.moveCall({
+        target: `${packageId}::bytes32::new`,
+        arguments: [tx.pure(arrayify(domainSeparator))],
+    });
+
+    tx.moveCall({
+        target: `${packageId}::gateway::setup`,
+        arguments: [
+            tx.object(creatorCap),
+            tx.pure.address(operator),
+            separator,
+            tx.pure.u64(minimumRotationDelay),
+            tx.pure.u64(options.previousSigners),
+            tx.pure(bcs.vector(bcs.u8()).serialize(encodedSigners).toBytes()),
+            tx.object(suiClockAddress),
+        ],
+    });
+
+    if (policy !== 'any_upgrade') {
+        const upgradeType = UPGRADE_POLICIES[policy];
```

Comment:
handle upgrade policy config for all contracts generically instead of just the gateway. An override for the owner for the upgrade cap can be passed in as well, along with the upgrade policy.